shanoir-ng-front/src/app/shared/components/entity/entity-list.component.abstract.ts

```diff
@@ -123,46 +123,38 @@ export abstract class EntityListComponent<T extends Entity> implements OnDestroy
         const dialogMsg : string = 'Are you sure you want to finally delete the ' + this.ROUTING_NAME
             + (entity['name'] ? ' "' + entity['name'] + '"' : ' with id n° ' + entity.id) + ' ?';
 
-        let promise: Promise<string>;
-        if (this.getOnDeleteConfirmMessage) {
-            promise = this.getOnDeleteConfirmMessage(entity);
-        } else {
-            promise = Promise.resolve('');
-        }
-        promise.then(studyListStr => {
-            this.confirmDialogService
-                .confirm(
-                    dialogTitle,
-                    dialogMsg + studyListStr
-                ).then(res => {
-                if (res) {
-                    this.getService().delete(entity.id).then(() => {
-                        this.onDelete.next({entity: entity});
-                        this.table.refresh().then(() => {
-                            if (this.ROUTING_NAME == 'examination') {
-                                this.consoleService.log('info', 'The ' + this.ROUTING_NAME + ' n°' + entity.id + ' has sucessfully started to delete. Check the job page to see its progress.');
-                            } else {
-                                this.consoleService.log('info', 'The ' + this.ROUTING_NAME + ' n°' + entity.id + ' sucessfully deleted');
-                            }
-                        });
-                        this.treeService.updateTree();
-                    }).catch(reason => {
-                        if (!reason){
-                            return;
-                        }
-                        if (reason instanceof ShanoirError && reason.code == 422) {
-                            this.dealWithDeleteError(reason, entity);
-                            return;
-                        } else if (reason.error){
-                            this.dealWithDeleteError(new ShanoirError(reason), entity);
-                            return;
+        const studyListStr = this.getOnDeleteConfirmMessage(entity);
```

Comment:
I believe that since this method has no default value, if undefined there will be an exception.
-> tested on studycard list > delete sc > error msg